packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -840,6 +844,91 @@ class ThemeData with Diagnosticable {
   factory ThemeData.dark({bool? useMaterial3}) =>
       ThemeData(brightness: Brightness.dark, useMaterial3: useMaterial3);
 
+  factory ThemeData._highContrast({
+    required bool useMaterial3,
+    required ColorScheme colorScheme,
+    required TextTheme textTheme,
+    SystemColorPalette? systemColors,
+  }) {
+    final ThemeData theme =
+        useMaterial3
+            ? ThemeData.from(
+              useMaterial3: useMaterial3,
+              colorScheme: ColorScheme.fromSeed(
+                seedColor: const Color(0xFF6750A4),
+                brightness: colorScheme.brightness,
+                contrastLevel: 1.0,
+                primary: colorScheme.primary,
+                onPrimary: colorScheme.onPrimary,
+                secondary: colorScheme.secondary,
+                onSecondary: colorScheme.onSecondary,
+                surface: colorScheme.surface,
+                onSurface: colorScheme.onSurface,
+                error: colorScheme.error,
+                onError: colorScheme.onError,
+              ),
+              // textTheme: textTheme, ???
+            )
+            : ThemeData.from(
+              useMaterial3: useMaterial3,
+              colorScheme: colorScheme,
+              textTheme: textTheme,
+            );
+
+    if (systemColors == null) {
+      return theme;
+    }
+
+    return theme.copyWith(
+      elevatedButtonTheme: ElevatedButtonThemeData(
+        style: ElevatedButton.styleFrom(
+          foregroundColor: systemColors.buttonText?.value,
+          backgroundColor: systemColors.buttonFace?.value,
+        ),
```

Comment:
We can also add `side: systemColors.buttonBorder == null ? null : BorderSide(color: systemColors.buttonBorder!.value),`

Reply:
Done